doc/Overview/Mvux/Advanced/Pagination.md

```diff
@@ -0,0 +1,398 @@
+|CurrentCount|This is the total number of items currently in the list.|
+|DesiredSize|The desired number of items for the current page, if any.<br/><br/>This is the desired number of items that the view requested to load.<br/>It's expected to be null only for the first page.<br/>Be aware that this might change between pages (especially is user resize the window), DO NOT use in `source.Skip(page.Index * page.DesiredSize).Take(page.DesiredSize)`.<br/>Prefer to use the `CurrentCount` property, e.g. `source.Skip(page.CurrentCount).Take(page.DesiredSize)`.|
+
+## Keyset pagination with a cursor
+
+There are several caveats in using `Skip` and `Take` (Offset pagination) with an arbitrary page size multiplied by the page number.
+
+- When we skip data records, the database might still have to process some the skipped records on its way to the desired ones.
+- If any updates have been applied to the records preceding the currently displayed page, and then the user moves to the next or previous page, there might be inconsistencies in showing the subsequent data, some of the entries might be skipped or shown twice.
+
+An alternative way to paginate data is by using a cursor that points to a specific record and take then number of desired records in a page onwards.  
+This is referred to as 'keyset pagination' or 'seek-based pagination'.
+
+To utilize this pagination style, MVUX provides another `ListFeed` factory overload, the `AsyncPaginatedByCursor`.
+
+The signature of this method is:
+
+```csharp
+public static IListFeed<T> AsyncPaginatedByCursor<TCursor>(TCursor firstPage, GetPage<TCursor, T> getPage)
+```
+
+where the `GetPage` type is a delegate with the following pattern:
+
+```
+public delegate ValueTask<PageResult<TCursor, TItem>> GetPage<TCursor, TItem>(TCursor cursor, uint? desiredPageSize, CancellationToken ct);
+```
+
+The callback delegate takes two parameters and returns two.
+
+The two input parameters are:
+
+- A cursor to the first page, in other words the item this page starts with. The service then starts by ordering the data and searching for the first item, then uses the second parameter:
+- The desired page size, the number of entries following it, along with another entity that uses as the cursor of the next page, which is not displayed to the user.
+- A `CancallationToken` which can be used to abort the operation.
+
+The callback returns a `PageResult<TCursor, TItem>`, and it contains:
+
+- An `ImmutableList<T>` containing the items to be displayed in the resulted page.
+- A `TCursor` that can be of a different type that points to the beginning of the upcoming page.
+
+To demonstrate this we'll stick to the *PaginatedPeopleApp* we've started before.
+
+### Service
+
+To keep the service agnostic, we'll add the following method to the Service:
+
+*IPeopleService*:
+
+```csharp
+ValueTask<(IImmutableList<Person> CurrentPage, Person NextCursor)> GetPeopleAsync(Person? cursor, uint pageSize, CancellationToken ct);
```

Comment:
Using the type of the items (a `Person` here) as cursor is not really a common pattern. Usually the API returns a dedicated key (e.g. a `string`) with the first page result, then we only have to pass it to fetch the next page.

For the demo here, you could use the id of the last person to not have the same type.